website/server/libs/slack.js

```diff
@@ -77,14 +89,75 @@ function sendFlagNotification ({
   });
 }
 
+function sendInboxFlagNotification ({
+  authorEmail,
+  flagger,
+  message,
+  userComment,
+}) {
+  if (SKIP_FLAG_METHODS) {
+    return;
+  }
+  let titleLink;
+  let authorName;
+  let title = `Flag in ${flagger.profile.name}'s Inbox`;
+  let text = `${flagger.profile.name} (${flagger.id}; language: ${flagger.preferences.language}) flagged a PM`;
+  let footer = `<${SLACK_FLAGGING_FOOTER_LINK}?userId=${flagger.id}&chatMessageId=${message.id}|Flag this message>`;
+
+  if (userComment) {
+    text += ` and commented: ${userComment}`;
+  }
+
+  if (!message.user && message.uuid === 'system') {
+    authorName = 'System Message';
+  } else {
+    authorName = `${message.user} - ${authorEmail} - ${message.uuid}`;
+  }
+
+  let messageText = message.text;
+
+  if (flagger.id === message.uuid) {
+    messageText += `: ${flagger.profile.name} is writing to itself.`;
```

Comment:
This isn't necessarily true. `message.uuid` will be the User ID of the recipient if `sent` is true, or of the sender otherwise, and the flagger could be flagging either message.

It is possible for a player to send themselves a private message and then to flag it, but the chances of that happening are small so I don't think it's worth having code to add a special message for that. It should be clear enough from the other data in the slack message anyway.

Just as a general note though in case it's useful elsewhere, "itself" should be "themself" (or "themselves" if you prefer) because "it" isn't used to refer to people even when gender is unknown. :)

Reply:
Ok! 😁 well thats why have reviews 👍  thanks for pointing this out, I thought that might by helpful^^ - gonna remove it.